examples/example_art.cpp

```diff
@@ -46,11 +46,15 @@ int main() {
   std::cerr << "Insert key 50 result: " << insert_result << '\n';
 
   // visitor for scans.
-  auto fn = [](const unodb::visitor<typename unodb::db::iterator>& v) {
-    const auto& val = v.get_value();
-    const std::string_view s(reinterpret_cast<const char*>(val.data()),
+  auto fn = [](const unodb::visitor<unodb::db<std::uint64_t>::iterator>& vis) {
+    const auto& key{vis.get_key()};
+    unodb::key_decoder dec{key};
```

Comment:
A similar comment here.  In the example, we should be thinking about modeling what we want people to be doing.

Calling decode here makes sense if you have a simple primitive key type.  But it does not always make sense for complex key types using the variable length key mechanism.  

Just something to think about.  Maybe even a different example for the use case of secondary indices, except that we do not support a fixed width type for the Value and to me that would be part of the ideal secondary index pattern.

Reply:
Right and I think we should already split this example. Or just here the decodable key example

Reply:
Right and I think we should already split this example. Or just here the decodable key example